chore: use fileinput in build image form

deboer-tim opened the pull request:
### What does this PR do?

Just some tech debt, use the FileInput component in the build image form.

### Screenshot / video of UI

Before:

<img width="595" alt="Screenshot 2024-07-03 at 4 22 27 PM" src="https://github.com/containers/podman-desktop/assets/19958075/b3a3aecb-6d9d-4ace-858d-e4a90f27ec3a">

After:

<img width="595" alt="Screenshot 2024-07-03 at 4 22 40 PM" src="https://github.com/containers/podman-desktop/assets/19958075/455e3073-368e-404a-b12f-6195d4ba4cfc">

### What issues does this PR fix or reference?

Part of #7938.

### How to test this PR?

Build an image.

- [x] Tests are covering the bug fix or the new feature

benoitf commented:
It looks like that the e2e tests need to be updated

feloy commented:
On Mac, I cannot select files named Containerfile or Dockerfile:

![build-image-openfile](https://github.com/containers/podman-desktop/assets/9973512/903436e9-948e-447c-b01e-cb51f0fd5742)

deboer-tim commented:
> It looks like that the e2e tests need to be updated

I had unintentionally renamed the directory placeholder,  the change was unnecessary so I removed it.

> On Mac, I cannot select files named Containerfile or Dockerfile:

Unfortunately the Electron file filter doesn't allow different capitalization or regex, so I've removed the filter as well.
